### DISCUSSION TITLE
Server no longer accepts connections after long period of inactivity

### DISCUSSION THREAD

**Discussion by dafzor** [lloesche/valheim-server-docker#85] [Q&A] [answered]
I'm running the container on docker for desktop and while it seems to run ok most of the time after a long period of inactivity (computer is left to idle) it will no longer accept connections.

Looking at the container log shows no abnormal output that I can tell.

Attaching a log were server didn't allow connection and was restarted at around 21h18. 
[_valheim-server_logs.txt](https://github.com/lloesche/valheim-server-docker/files/6025292/_valheim-server_logs.txt)

edit: forgot to mention that i've read previous issues that indicated resource starvation could be a cause of problems so monitoring memory and cpu usage on the server.

**Comment by lloesche** [comment]
No concrete idea. My thoughts go in the direction of a Docker for Desktop UDP forwarding issue. But that's just a rough guess. Is this on WSL2 or Hyper-V?

Since there are no errors in the log and we know with some certainty this issue doesn't happen on Linux (our server is sometimes idle for days at a time as we mostly play on weekends, plus no other reports with more than 1M pulls) I would assume it safe to say it's unrelated to the Valheim server itself. Leaving the environment it's running in.

Docker for Desktop is really just Docker on Linux in a VM running on Windows so there's quite some networking magic going on behind the scenes to make this all work transparently. And judging by a quick search for Docker on Windows UDP issues it seems fair to say that UDP forwarding is not the most extensively tested feature.

I would try switching backends (WSL2 vs Hyper-V) and see if that makes any difference.

**Comment by dafzor** [comment]
Currently running on wsl2, I assumed it was my environment has I had not seen any similar reports going thought the existing issues.

That said I plan on moving to a linux VM on a dedicated virtualization host (running on my desktop was a temporary solution), but still wanted to report it in case someone else had the same problem or an obvious solution was available.

Thank you for your reply

**Comment by lloesche** [comment]
Good thinking, thanks for the report! I'll convert it to a Q&A discussion. Maybe somebody stumbles across a solution.

**Comment by dafzor** [comment, marked as answer]
I have moved the container to a Debian VM in virtual box on the same host and the connection issues didn't happen.

So it all leads me to conclude that using Docker for Windows with wsl2 backend (the default) was the cause of the connection issues.

Might be worth putting a note in the readme to clarify Docker for Windows is not a viable option to host this container.
